lsp-mode.el

```diff
@@ -3743,7 +3729,7 @@ yet."
   (cond
    (lsp-managed-mode
     (when (lsp-feature? "textDocument/hover")
-      (add-function :before-until (local 'eldoc-documentation-function) #'lsp-eldoc-function)
+      (add-hook 'eldoc-documentation-functions #'lsp-eldoc-function nil t)
```

Comment:
The `eldoc-documentation-functions` seems to be the new one introduced at emacs 28 right?
You will probably need to check for its availability before use and fall back to the previous one since we support from emacs 26

Reply:
@kiennq 

Looks to me like the hook function was added in commit [c0fcbd2c119b8418855f0931aceefbef717c5e53](https://github.com/emacs-mirror/emacs/commit/c0fcbd2c119b8418855f0931aceefbef717c5e53) meaning it should've been in the 26.2 release. Plus @yyoncho suggested downloading a newer version of eldoc from MELPA for use with lsp-mode so I don't see much value in trying to support both hooks and the old add-function mechanism.